MAINTAINERS.md

```diff
@@ -1,5 +1,97 @@
+non-maintainers cannot. Being an OPI maintainer represents a significant level
+of trust to the broader project. Maintainer rights are not granted lightly,
+and will be reviewed as necessary to ensure that trust has not been violated.
+
+### Granting Maintainer Rights
+
+Granting maintainer rights will be considered when a candidate has demonstrated
+the following in their interation with the project:
+
+* Contribution of significant new features to OPI.
+  * Submissions are free of obvious defects.
+  * Submissions do not require many revisions.
+* Consistent and meaningful code reviews of other people's pull requests.
+* Helping others in the community via channels such as the mailing list,
+  GitHub. Issues and PRs, and the Slack channels
+* The individual should have plans for sustained contribution to the project.
+
+The process to grant someone the maintainer role is simple:
+
+* An existing maintainer nominates someone by sending a message to the
+  maintainers slack channel.
+* Existing maintainers discuss the idea of giving the candidate maintainer
+  rights.
+* After discussion has ended, existing maintainers will vote on approving the
+  new maintainer.
+* To recieve maintainership, the candidate must receive a majority of yes votes
+  and zero no votes.
+* One final discussion with the candidate happens to ensure the candidate is
+  ready to accept the maintainer role.
+* If successful, the candidate is added to the appropriate GitHub group to
+  gain maintainer access.
+
+### Revoking Maintainer Rights
+
+When a maintainer acts in a way which is either detrimental to the OPI project,
+it raises a situation which needs to be handled quickly to ensure the long-term
+success and stability of the project. It is possible the maintainer in question
+will have their maintainer rights removed. The process for this is as follows:
+
+* Discuss with the individual privately about the concerns. Use email for this,
+  sticking to the facts, and try to reach a resolution on the individuals.
+  behavior. Ensure the individual understands their behavior needs to change.
+* Send an email discussing the individuals behavior to the maintainer mailing
+  list. Work to resolve the issue in this forum.
+* If necessary, start a thread with maintainers except the individual in
+  question regarding revoking the individuals maintainer rights.
+* Each maintainer may vote yes, no, or abstain. Not voting is equivalent to
+  abstain.
+* After all votes are collected, to remove the maintainer, the vote must be
+  a 2/3 yes vote in favor of removal.
+* Any no votes should document their reasons for not voting to remove the
+  persons maintainership.
+* The person proposing revoking maintainer rights sends email to the list
+  announcing the results.
+* If the vote to remove is successful, the candidate will be removed from the
+  GitHub group which has merge rights.
+* Ideally, the revoked committer peacefully leaves the community. If this is
```

Comment:
Is the intent to have a revoked maintainer leave the community and stop contributing or simply to revoke the maintainer role? I get that we don't want someone who disagrees with having the maintainer role revoked being a bad apple in the community, but I was happy to renounce my maintainer right because I do not yet have the GH savvy and other qualifications to be a maintainer, but I am still wanting to be a contributor. As an aside should the word "committer"  be "maintainer" .

Reply:
I think in general, the revocation process will be used for folks who may not want their rights revoked. In your case Paul, you were a peaceful ex-maintainer. I can make it more clear that this part of the process is for non-peaceful departures. Hopefully we have none, but history suggests it will happen.